src/ipfs/config.ts

```diff
@@ -13,11 +14,9 @@ export const setIpfs = (userIpfs: unknown): void => {
   ipfs = userIpfs as IPFS
 }
 
-export const getIpfs = async (): Promise<IPFS> => {
-  if (ipfs) {
-    return ipfs
-  }
-  ipfs = (await getIpfsWithCfg(IPFS_CFG)) as unknown as IPFS
+export const getIpfs = async (options = {}): Promise<IPFS> => {
+  if (ipfs) return ipfs
+  ipfs = (await getIpfsWithCfg({ ...IPFS_CFG, ...options })) as unknown as IPFS
```

Comment:
I'm using the following in Drive now to setup IPFS:
```js
  await sdk.ipfs.getIpfs({
    permissions: [
      "cat",
      "dag.tree",
      "files.catPullStream",
      "files.catReadableStream",
      "id",
      "ls",
      "swarm.connect",
      "version",
    ],

    jsIpfs: "./web_modules/ipfs.min.js"
  })
```
@dholms Do you think that's fine? I'm a bit worried that `getIpfs` will be called on "script load" and my options won't go through, but I think that's not the case?

Reply:
Yes this should be fine :+1: 

Two notes:
- permissions are only necessary for ipfs companion. the js-ipfs daemon is automatically permissioned (since it's sandboxed by the browser)
- `ipfs` will be set by whatever calls `getIpfs` first. So if you want a custom config passed, just make sure you call `sdk.ipfs.getIpfs` before doing any filesystem actions/ anything from the sdk that uses ipfs